src/plugins/mf_priority.cpp

```diff
@@ -979,8 +979,18 @@ static int inactive_cb (flux_plugin_t *p,
         return -1;
     }
 
-    b->cur_run_jobs--;
-    b->cur_active_jobs--;
+    // if the job posted an "alloc" event, then we know the job actually ran,
+    // so we need to decrement both the current running jobs count and the
+    // current active jobs count; if not, then we only need to decrement the
+    // current active jobs count and exit this callback.
+    if (flux_jobtap_job_event_posted (p, FLUX_JOBTAP_CURRENT_JOB, "alloc")) {
+        b->cur_run_jobs--;
+        b->cur_active_jobs--;
+    } else {
+        b->cur_active_jobs--;
+
+        return 0;
+    }
```

Comment:
IMO, this logic would be clearer by

 1. first decrement active jobs count, this makes it clear that count is always decremented.
 2. check if no `alloc` event posted, then nothing more to do, so return
 3. proceed to decrement `cur_run_jobs` and check for any held jobs that need to be released

i.e.:
```c
    b->cur_active_jobs--;
    // Nothing more to do if this job was never running
    if (!flux_jobtap_job_event_posted (p, FLUX_JOBTAP_CURRENT_JOB, "alloc"))
        return 0;

    // This job was running, decrement running jobs count and check if a job can be released
    b->cur_run_jobs--;

```